recipes/xapian-core/all/conanfile.py

```diff
@@ -1,14 +1,18 @@
 from conan import ConanFile
-from conan.tools.files import rename, apply_conandata_patches, replace_in_file, rmdir, save, rm, get
-from conan.tools.microsoft import is_msvc
-from conan.tools.scm import Version
 from conan.errors import ConanInvalidConfiguration
-from conans import AutoToolsBuildEnvironment, tools
-from contextlib import contextmanager
-import functools
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import cross_building
+from conan.tools.env import Environment, VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rename, rm, rmdir, save
+from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc, unix_path
+from conan.tools.scm import Version
+from conans import tools as tools_legacy
```

Comment:
This isn't conan 2 compatible, is it? Looking through the root of `conan.conans` on develop2, I can't see a `tools` module or subpackage. It looks like you only need this for `unix_path` though, so why not use `conan.tools.microsoft.unix_path` as you've already imported?

Reply:
because new unix_path doesn't make sense in package_info(), it's no-op.

Reply:
@jcar87 this env var interacts with automake recipe in case someone needs to expose ACLOCAL of xapian-core. How to achieve proper interaction in conan v2 on Windows without legacy unix_path()?

Reply:
So, after a bit of experimentation, it looks like it doesn't matter where in the recipe `conan.tools.microsoft.unix_path` is, it just needs to meet a few conditions in order not to be a no-op:

* Scope indpendent, the following are defined:
    * `tools.microsoft.bash:subsystem={msys,msys2,cygwin,wsl,sfu}`
    * `tools.microsoft.bash:active=True`
* `unix_path(conanfile, path, scope="build")`
    *  `tools.microsoft.bash:subsystem={msys,msys2,cygwin,wsl,sfu}`
    * `conanfile.win_bash=True`
* `unix_path(conanfile, path, scope="run")`
    *  `tools.microsoft.bash:subsystem={msys,msys2,cygwin,wsl,sfu}`
    * `conanfile.win_bash_run=True`

@SpaceIm are you able to elaborate why it would be a no-op in the package_info method but not elsewhere?

I've tested using it in package_info and it appears to work there. Maybe for this case, we need to use the `run` scope since it defaults to build, so

```diff
- self.env_info.AUTOMAKE_CONAN_INCLUDES.append(tools_legacy.unix_path(os.path.join(self._datarootdir, "aclocal")))
+ self.env_info.AUTOMAKE_CONAN_INCLUDES.append(unix_path(self, os.path.join(self._datarootdir, "aclocal"), scope="run"))
```
but also define 
```python
class XapianCoreConan(ConanFile):
    ...
    win_bash_run=True
    win_bash=True
```

Reply:
Speculating - it's a no-op in package_info when `scope="build"`. I didn't test that.

Reply:
https://github.com/conan-io/conan-center-index/pull/14792/commits/5c8721960fbcc7b04e77cbdb8832affdebbf7b14 should help, it's the same workaround used in libtool recipe.